File: app/src/main/java/com/example/paetz/yacguide/database/Comment/RegionComment.java
package com.example.paetz.yacguide.database.Comment;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;

import java.util.HashMap;
import java.util.Map;

@Entity
public class RegionComment {

    // This needs to be in sync with sandsteinklettern.de!
    public static final Map<Integer, String> QUALITY_MAP = new HashMap<Integer, String>() {{
        put(1, "international bedeutend");
        put(2, "national bedeutend");
        put(3, "lohnt einen Abstecher");
        put(4, "lokal bedeutend");
        put(5, "unbedeutend");
    }};

    @PrimaryKey
    private int id;

    private int qualityId;
    private String text;
    private int regionId;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getQualityId() {
        return qualityId;
    }

    public void setQualityId(int qualityId) {
        this.qualityId = qualityId;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public int getRegionId() {
        return regionId;
    }

    public void setRegionId(int regionId) {
        this.regionId = regionId;
    }
}

File: app/src/main/java/com/yacgroup/yacguide/database/Comment/RegionComment.kt
/*
 * Copyright (C) 2019 Fabian Kantereit
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package com.yacgroup.yacguide.database.Comment

import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey

import java.util.HashMap

@Entity
class RegionComment {

    @PrimaryKey
    var id: Int = 0

    var qualityId: Int = 0
    var text: String? = null
    var regionId: Int = 0

    companion object {

        const val SELECT_ALL = "SELECT RegionComment.* FROM RegionComment"
        const val DELETE_ALL = "DELETE FROM RegionComment"
        const val FOR_REGION = "WHERE RegionComment.regionId ="

        // This needs to be in sync with sandsteinklettern.de!
        val QUALITY_MAP: Map<Int, String> = object : HashMap<Int, String>() {
            init {
                put(1, "international bedeutend")
                put(2, "national bedeutend")
                put(3, "lohnt einen Abstecher")
                put(4, "lokal bedeutend")
                put(5, "unbedeutend")
            }
        }
    }
}
